### PR TITLE
sample: buck

### PR DESCRIPTION
**Pull request by wilu-nordic**

### CODE REVIEW

File: samples/buck/README.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
+.. _npmx_buck_sample:
```
Comment (masz-nordic) [inline samples/buck/README.rst:1]:
```suggestion
.. _buck_sample:
```

File: samples/buck/README.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+BUCK
+####
+
+.. contents::
+   :local:
+   :depth: 2
+
+The npmx BUCK sample demonstrates how to control PMIC BUCKs using npmx drivers.
```
Comment (masz-nordic) [inline samples/buck/README.rst:10]:
```suggestion
The BUCK sample demonstrates how to control PMIC BUCKs using npmx drivers.
```

File: samples/buck/src/main.c [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+#include <npmx_gpio.h>
+
+#define LOG_MODULE_NAME pmic_buck
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/* Values used to refer to the specified buck. */
+#define BUCK1_IDX 0
+#define BUCK2_IDX 1
+
+#if CONFIG_TESTCASE_SET_BUCK_VOLTAGE || CONFIG_TESTCASE_OUTPUT_VOLTAGE ||                          \
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply samples/buck/src/main.c:20]: `\` needs to be at the end of 100 character line. Forced by .clang-format.

**Review by annakielar — DISMISSED** [review]

**Review by mkleszcz-no — DISMISSED** [review]

File: samples/buck/sample.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+common:
+  integration_platforms:
+    - nrf5340dk_nrf5340_cpuapp
+    - nrf52840dk_nrf52840
+  platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840
+  tags: pmic
+  timeout: 30
+
+tests:
+  sample.pmic.buck.set_buck_voltage_eng_b:
```
Comment (masz-nordic) [inline samples/buck/sample.yaml:14]:
```suggestion
  sample.buck.set_buck_voltage_eng_b:
```
Same in all instances below.

Comment (wilu-nordic) [reply samples/buck/sample.yaml]: Fixed

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**